charts/scalardb/templates/scalardb/deployment.yaml

```diff
@@ -47,8 +46,34 @@ spec:
             - containerPort: 50051
             - containerPort: 8080
           env:
-          - name: JAVA_OPTS
-            value: "-Dlog4j.logLevel={{ .Values.scalardb.storageConfiguration.dbLogLevel }}"
+          - name: SCALAR_DB_CONTACT_POINTS
+            value: "{{ .Values.scalardb.storageConfiguration.contactPoints }}"
+          - name: SCALAR_DB_CONTACT_PORT
+            value: "{{ .Values.scalardb.storageConfiguration.contactPort }}"
+          - name: SCALAR_DB_USERNAME
+            valueFrom:
+              secretKeyRef:
+              {{- if .Values.scalardb.existingSecret }}
+                name: {{ .Values.scalardb.existingSecret }}
+              {{- else }}
+                name: {{ include "scalardb.fullname" . }}-secret
+              {{- end }}
+                key: db-username
+          - name: SCALAR_DB_PASSWORD
+            valueFrom:
+              secretKeyRef:
+              {{- if .Values.scalardb.existingSecret }}
+                name: {{ .Values.scalardb.existingSecret }}
+              {{- else }}
+                name: {{ include "scalardb.fullname" . }}-secret
+              {{- end }}
+                key: db-password
+          - name: SCALAR_DB_STORAGE
+            value: "{{ .Values.scalardb.storageConfiguration.storage }}"
+          - name: SCALAR_DB_LOG_LEVEL
+            value: "{{ .Values.scalardb.storageConfiguration.dbLogLevel }}"
+          - name: SCALAR_DB_SERVER_PORT
+            value: "50051"
```

Comment:
In the current chart, the value of `scalar.db.server.port` is hard-coded as `50051`.
This port is only used internal connections and is not exposed to user to access Scalar DB Server.
Basically, uses accesss Scalar DB Server via Service resource (Envoy).

So, I think that this hard-coding is not critical issue and we don't need fix now.
However, if we need to fix it in the future, I will do that in the another PR.

Reply:
I have just noticed that the default port of the Scalar DB Server is 60051, isn't it?
And if we use the default port, perhaps is it possible to leave the value blank in this yaml file?

Reply:
Yes. The default port of Scalar DB Server is 60051.
I think this configuration (hard-coded 50051) related to the Envoy. 
It's just a guess, but I think the default values of Scalar Envoy Charts (50051 as target port) is set based on the assumption of Scalar DL.

So, I think we can use 60051 port and remove hard-coded value, if I we set target port of Envoy to 60051 and fix several hard-coded point (50051) in the YAML file.

However, I didn't test the above configuration yet.
So, I will try to test using above configurations today.
Could you please give me some time?

Reply:
Got it! Let's keep the port 50051 for now. Thank you for confirming!

Reply:
For your reference.

I checked the configurations of Scalar Envoy, Scalar Envoy Charts, and Scalar DB Charts.
As a result, I found the way to remove this hard-coded value.
I need to change the `deployment.yaml` and to use custom values file as follows.

* File changed
  ```diff
  --- a/charts/scalardb/templates/scalardb/deployment.yaml
  +++ b/charts/scalardb/templates/scalardb/deployment.yaml
  @@ -43,7 +43,7 @@ spec:
             image: "{{ .Values.scalardb.image.repository }}:{{ .Values.scalardb.image.tag | default .Chart.AppVersion }}"
             imagePullPolicy: {{ .Values.scalardb.image.pullPolicy }}
             ports:
  -            - containerPort: 50051
  +            - containerPort: 60051
               - containerPort: 8080
             env:
             - name: SCALAR_DB_CONTACT_POINTS
  @@ -72,15 +72,13 @@ spec:
               value: "{{ .Values.scalardb.storageConfiguration.storage }}"
             - name: SCALAR_DB_LOG_LEVEL
               value: "{{ .Values.scalardb.storageConfiguration.dbLogLevel }}"
  -          - name: SCALAR_DB_SERVER_PORT
  -            value: "50051"
             resources:
               {{- toYaml .Values.scalardb.resources | nindent 12 }}
             livenessProbe:
               exec:
                 command:
                 - /usr/local/bin/grpc_health_probe
  -              - -addr=:50051
  +              - -addr=:60051
               failureThreshold: 3
               initialDelaySeconds: 10
               periodSeconds: 10
  ```
* Custom values file
  ```yaml
  envoy:
    envoyConfiguration:
      serviceListeners: "scalardb-service:60051"
    service:
      ports: 
        envoy:
          port: 60051
          targetPort: 60051
      type: "LoadBalancer" # For testing. This is not necessary.
  scalardb:
    image:
      repository: "local/test/scalardb-server"
      tag: "4.0.0-SNAPSHOT"
    service:
      ports: 
        scalardb:
          port: 60051
          targetPort: 60051
  ```

When I apply the above changes and configuration, I was able to run Scalar DB Server with default port (60051).
I think we should change the default values (values.yaml file), if we want to remove this hard-coded value.

However, it works well for now, and I think this changes is not necessary for running Scalar DB Server using Helm Charts.
So, I will discuss and consider to fix this hard-coded value as an another task in the future.